### PR TITLE
insulate oncokb-annotator from dmp runs

### PR DESCRIPTION
**Pull request by sheridancbio**
- wait until dmp import run is complete before starting
- copy needed source files to a staging directory
- use a separate working directory rather than working in repo subdirectory
use variable names for file paths
repo cleanup not needed because repo is now unaltered

### CODE REVIEW

**Comment by sheridancbio** [comment]
One thing I wonder ... should we wrap this script with flock protection so that it cannot interfere with itself?

**Comment by ao508** [comment]
@sheridancbio I think it would be a nice preventative measure just in case

**Comment by sheridancbio** [comment]
I will add the flock protection. While testing I also added a timestamp for "last updated" to the generated index.html download page.

File: import-scripts/create_web_directory_index_file.sh [hunk 1]
```diff
@@ -38,6 +39,8 @@ function make_directory_file() {
         fi
     done < $FILELIST_FILENAME
     echo "</ul>" >> $INDEX_FILENAME
+    timestamp=$(date '+%Y_%m_%d %H:%M %Z')
+    echo "<p style=\"color:Teal;font-size:7px;\">(last updated: $timestamp)" >> $INDEX_FILENAME
```
Comment (mandawilson) [inline import-scripts/create_web_directory_index_file.sh:43]: Good.

File: import-scripts/oncokb-annotator.sh [hunk 2]
```diff
@@ -1,174 +1,298 @@
...
+            ONCOKB_ANNOTATION_SUCCESS=0
+        fi
+    fi
+
+    # Update DMP repo before running annotations
+    if [ $ONCOKB_ANNOTATION_SUCCESS -eq 1 ] ; then
+        echo $(date)
+        echo "Updating the DMP source repository..."
+        fetch_updates_in_data_sources dmp
+        if [ ${#failed_data_source_fetches[*]} -ne 0 ] ; then
```
Comment (mandawilson) [inline import-scripts/oncokb-annotator.sh:158]: `failed_data_source_fetches` set in `set-data-source-environment-vars.sh` and will contain a list of all data source fetches which failed after calling fetch_updates_in_data_sources()

File: import-scripts/oncokb-annotator.sh [hunk 3]
```diff
@@ -1,174 +1,298 @@
...
+        echo $(date)
+        echo "Beginning README generation..."
+        $PYTHON_BINARY $README_SCRIPT -o $ONCOKB_README_FILE -u $ONCOKB_URL
+        if [ $? -ne 0 ] ; then
+            echo "Failed to generate README, exiting..."
+            ONCOKB_ANNOTATION_SUCCESS=0
+        fi
+    fi
+
+    # Only commit if all steps succeeded
```
Comment (mandawilson) [inline import-scripts/oncokb-annotator.sh:278]: Only rsync if all steps succeeded (instead of commit?)

Comment (sheridancbio) [reply import-scripts/oncokb-annotator.sh:278]: right .. there is no modification of the repository here. We upload the annotated oncokb files to ramen as a community download.

**Pull request merged**